main/adapters/repository/protocol_test.go

```diff
@@ -117,6 +123,19 @@ func TestExtendedProtocol_LoadPublicKey(t *testing.T) {
 	assert.Equal(t, i.PublicKey, pub)
 }
 
+func TestExtendedProtocol_LoadPublicKeyFromExternalIdentity(t *testing.T) {
+	extId := ent.ExternalIdentity{Uid: testUid, PublicKey: testPubKeyBytes}
+
+	p, err := NewExtendedProtocol(&MockCtxMngr{
```

Comment:
I may be missing some test function elsewhere, but since we are using only the mock context, and the mock context implements, well, only mocks of the database access function (see https://github.com/ubirch/ubirch-client-go/pull/103/files#diff-68338c1ad2792764901663fae6a82c22b72558c1532448b5d4c178617a7acd8aR140 and below) we might only be testing the implementation of the mock and never the actual implementation. My coverage analysis unfortunately does not work at the moment but maybe we can check if we actually test the sqlite access code for external identities.

Reply:
the actual database operations are tested here [sqlite](https://github.com/ubirch/ubirch-client-go/pull/103/files#diff-08d5bc978de1baf88790c2cb3c0226e3c4a8b173c33de2d5082ca19b77a8ab1eR528) and here [postgres](https://github.com/ubirch/ubirch-client-go/pull/103/files#diff-5fcdd0a18d28dbf07caf122a3a3ac6f99f2e3c19039b3ee9b0b5252b8599e7c1R510)